### PR TITLE
spfft: add version 1.1.0

### PR DESCRIPTION
**Pull request by AdhocMan**
Add SpFFT version 1.1.0 and update version requirements of dependencies.

### CODE REVIEW

File: var/spack/repos/builtin/packages/spfft/package.py [hunk 1]
```diff
@@ -50,15 +51,22 @@ class Spfft(CMakePackage, CudaPackage, ROCmPackage):
...
-    depends_on("cuda@:10", when="@:0.9.11 +cuda")
+    depends_on("cuda@9:10", when="@:0.9.11 +cuda")
+    depends_on("cuda@9:", when="@0.9.12:1.0.6 +cuda")
+    depends_on("cuda@11:", when="@1.1.0: +cuda")
 
     # Workaround for compiler bug in ROCm 4.5+ added in SpFFT 1.0.6
     conflicts("+rocm", when="@:1.0.5")
 
     with when("+rocm"):
+        depends_on("hip +rocm ~cuda")
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/spfft/package.py]: Note this is redundant given constraints in the `hip` package.

File: var/spack/repos/builtin/packages/spfft/package.py [hunk 2]
```diff
@@ -50,15 +51,22 @@ class Spfft(CMakePackage, CudaPackage, ROCmPackage):
...
         depends_on("rocfft")
-        depends_on("hipfft")
+        depends_on("hipfft +rocm ~cuda")
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/spfft/package.py]: Note this is redundant given constraints in the `hipfft` package.

**Review by scheibelp — COMMENTED** [review]
Can you remove the redundant constraints related to avoiding both +rocm and +cuda on `hip`/`hipfft`?

All the other edits here look good to me.

**Review by bernhardkaindl — APPROVED** [review]
Thanks @AdhocMan! I've seen that the changes requested by scheibelp are pushed!

**Pull request merged**